helios-services/src/main/java/com/spotify/helios/servicescommon/coordination/DefaultZooKeeperClient.java

```diff
@@ -159,7 +165,9 @@ public Stat exists(final String path) throws KeeperException {
 
   @Override
   public void start() {
-    client.start();
+    if (client.getState() != CuratorFrameworkState.STARTED) {
```

Comment:
is `start()` really called more than once? 

If it is - shouldn't the lines below for adding connectionStateListener etc also be in this block?